### PR TITLE
Fix decoding uppercase letters, special keys. US key board.

### PR DESCRIPTION
**Pull request by aragats**
I would like to suggest this little fix.

Bug: decodeKey function worked in wrong way for special keys (~!@# and so on). and for lower case letters.

This little change is based on the follow principle:
1. CAPS_LOCK affects only letters. 
2. if CAPS_LOCK is true, then it affects letters as follows: lower case letters become uppercase and uppercase letters become lowercase
3. key.isupper() returns true only for uppercase letters (AQWEDF). For special keys, it returns false.
4. There are special keys, which we can type without holding SHIFT, and there are which require holding SHIFT. As I mentioned,  key.isupper()  does not return true for special keys typed with holding key. That's why we should have a list of special keys which must be typed with holding SHIFT.  Of course, it depends on keyboard layout, but I imply the US keyboard layout. The adjustment for other keyboards could be done later, if it will be required.

### CODE REVIEW

File: vncdotool/client.py [hunk 1]
```diff
@@ -131,7 +133,9 @@ class VNCDoToolClient(rfb.RFBClient):
     cmask = None
 
     def _decodeKey(self, key):
-        if self.factory.force_caps and key.isupper():
+        if (self.factory.force_caps and key.isalpha() and key.islower()) \
```
Comment (sibson) [inline vncdotool/client.py]: is it possible to send a lower case a when force_caps is true?

Comment (aragats) [inline vncdotool/client.py]: No, when self.factory.force_caps is true and you send lower case letter then it becomes upper case letter. It is like when you are typing with switched on CAPS LOCK.

Comment (sibson) [inline vncdotool/client.py]: ok, I like points 3 and 4, with the aim of fixing special keys #26 but I don't understand why the CAPS LOCK case inversion behaviour is desirable.  --force-caps is intended to work around vncservers that aren't properly handling upper case letters.

**Comment by sibson** [comment]
--force-caps is not intended to simulate caps-lock.  Rather, it forces [A-Z] to be prepended with shift. As some servers treat [A-Z] the same as [a-z].   I'm not convinced making it behave like caps-lock is desirable.  Are there use cases where you need cap-lock like behaviour?

**Pull request closed without merging**